packages/taco/src/conditions/schemas/context.ts

```diff
@@ -6,7 +6,8 @@ import { plainStringSchema } from './common';
 
 export const contextParamSchema = z.string().regex(CONTEXT_PARAM_REGEXP);
 
-const paramSchema = z.union([plainStringSchema, z.boolean(), z.number()]);
+// TODO: This is too broad, but it's a start
+const paramSchema = z.union([plainStringSchema, z.boolean(), z.number().int().nonnegative()]);
```

Comment:
Is there ever a case for negative numbers?

Reply:
You're right, it is indeed acceptable (`int256` type and friends). Fixed in b3902d4fadeaabec87b754c57e18719a17f3c3ce

Reply:
I was thinking about this some more. Perhaps we need to separate this schema into two parts.

1. What is allowed for blockchain RPC / smart contract calls
2. What is generally allowed

The original goal of the `paraSchema` on L9 was 2), but we are now limiting it to 1).

Solidity doesn't have floating point numbers, so limiting it to `z.number().int()` is fine. However, a `JsonApiCondition` can require floating point numbers. For example, the `paramOrContextParamSchema` is used for the value used for the returnValueTest () - and for a JSON API, checking a floating point number is perfectly fine, but now it won't be (I saw you had to modify the floating point test)

Reply:
^ @cygnusv wdyt?

Reply:
yep, I was actually reverting my changes as we speak. Let's just put a pin on this so we can unblock the PR.

Reply:
Opened https://github.com/nucypher/taco-web/issues/600